components/script/dom/window.rs

```diff
@@ -1452,7 +1452,7 @@ impl Window {
 
     #[allow(unsafe_code)]
     pub fn offset_parent_query(&self, node: TrustedNodeAddress) -> (Option<DomRoot<Element>>, Rect<Au>) {
-        if !self.reflow(ReflowGoal::OffsetParentQuery(node), ReflowReason::Query) {
+        if !self.reflow(ReflowGoal::QueryLatency(QueryMsg::OffsetParentQuery(node), u64::default()), ReflowReason::Query) {
```

Comment:
For all of these cases, I recommend creating a `layout_query` method that accepts a QueryMsg argument and calling that instead of `reflow`. That will avoid duplicating all of the code that creates a time value, and it won't be necessary to pass a ReflowGoal argument.

Reply:
thank for your advice, but I'm a little puzzled by the function of `layout_reflow`. if it's mainly same as `reflow`, the `layout_reflow` method seems superfluous.
So maybe I can place a closure inside every `xx_xx_query` method like this: 
```
let reflow_goal = |node| { 
     ReflowGoal::LayoutQuery(QueryMsg::XxXxQuery(node), u64::default()
}
``` 
But this looks untidy too. Could you make the function of `layout_query` more explicit to me?
(sorry for bothering you)

Reply:
I'm imagining something like this:
```rust
fn layout_query(query: QueryMsg) {
    self.reflow(ReflowGoal::LayoutQuery(query), u64::default())
}
```
Where u64::default() will be replaced with whatever time-initialization code makes sense.